backend/src/worker.rs

```diff
@@ -1536,18 +1599,22 @@ async fn set_logs(logs: &str, id: uuid::Uuid, db: &DB) {
     };
 }
 
-async fn concat_logs(logs: &str, id: &Uuid, db: &DB) {
-    if sqlx::query!(
+/* TODO retry this? */
+async fn append_logs(job_id: uuid::Uuid, logs: impl AsRef<str>, db: impl Borrow<DB>) {
+    if logs.as_ref().is_empty() {
+        return;
+    }
+
+    if let Err(err) = sqlx::query!(
         "UPDATE queue SET logs = concat(logs, $1::text) WHERE id = $2",
-        logs.to_owned(),
-        id
+        logs.as_ref(),
+        job_id,
     )
-    .execute(db)
+    .execute(db.borrow())
     .await
-    .is_err()
     {
-        tracing::error!("error updating logs for id {}", id)
```

Comment:
For now we should keep critical information like id the message itself even though it's good to keep it as a key. That is because the log pipeline up to grafana does not always keep the metadata but always keep the message. Let's revisit this when we do a cycle on logging.